subscription-service/src/main/kotlin/com/egm/stellio/subscription/web/SubscriptionHandler.kt

```diff
@@ -135,15 +141,45 @@ class SubscriptionHandler(
     fun delete(req: ServerRequest): Mono<ServerResponse> {
         val subscriptionId = req.pathVariable("subscriptionId")
 
-        return extractJwT()
-                .flatMap {
-                    subscriptionService.delete(subscriptionId, it.subject)
-                }
-                .flatMap {
-                    if (it >= 1)
-                        noContent().build()
-                    else
-                        notFound().build()
-                }
+        return checkSubscriptionExists(subscriptionId)
+            .flatMap {
+                extractJwT()
+            }
+            .flatMap {
+                checkIsAllowed(subscriptionId, it.subject)
+            }
+            .flatMap {
+                subscriptionService.delete(subscriptionId)
+            }
+            .flatMap {
+                noContent().build()
+            }
     }
+
+    private fun checkSubscriptionExists(subscriptionId: String): Mono<String> =
+        subscriptionService.exists(subscriptionId)
+            .flatMap {
+                if (!it)
+                    Mono.error(ResourceNotFoundException("Could not find a subscription with id $subscriptionId"))
+                else
+                    Mono.just(subscriptionId)
+            }
+
+    private fun isNewSubscription(subscriptionId: String): Mono<String> =
```

Comment:
The `checkXXX` wording more clearly defines the intent of a function.

When a function is named `isXXX`, I expect it to return a boolean ...